R/clean_data_helpers.R

```diff
@@ -1,76 +1,147 @@
+  # the numeric values.
+  #     When there is no Date values identified from the parsing, the variable
+  # is converted into numeric. The numeric count is the sum of numeric values.
+  #     The logical count is the number of TRUE and FALSE written in both lower
+  # and upper case within the variable
+  #     The remaining values will be of type character.
 
-  # --- get the proportion of date values ---
-  x        <- x[which(is.na(tmp))]
-  are_date <- 0L
-  if (!is.null(lubridate::guess_formats(x, c("ymd", "ydm", "dmy", "mdy", "myd",
-                                             "dym", "Ymd", "Ydm", "dmY", "mdY",
-                                             "mYd", "dYm")))) {
-    x <- suppressWarnings(
-      as.Date(
-        lubridate::parse_date_time(
-          x, orders = c("ymd", "ydm", "dmy", "mdy", "myd", "dym", "Ymd", "Ydm",
-                        "dmY", "mdY", "mYd", "dYm")
-        )
+  # parsing the vector, looking for date values
+  tmp <- suppressWarnings(
+    as.Date(
+      lubridate::parse_date_time(
+        x,
+        orders = c("ymd", "ydm", "dmy", "mdy", "myd", "dym", "Ymd", "Ydm",
+                   "dmY", "mdY", "mYd", "dYm")
       )
     )
-    are_date <- round((sum(!is.na(x)) / n_rows), 6L)
+  )
+
+  # getting the date and numeric count as describe above
+  date_count <- numeric_count <- 0L
+  if (sum(!is.na(tmp)) > 0L) {
+    # Setting the first date to 50 years before the current date
+    target_interval <- sort(
+      seq.Date(Sys.Date(), length.out = 2L, by = "-50 years")
+    )
+
+    # get the date count
+    date_count <- date_count + sum(!is.na(tmp))
+
+    # convert to numeric and check for the presence of Date among the numeric
+    tmp2 <- x[is.na(tmp)]
+    tmp3 <- suppressWarnings(as.numeric(tmp2))
+    if (sum(!is.na(tmp3)) > 0L) {
+      y <- lubridate::as_date(
+        tmp3[!is.na(tmp3)],
+        origin = target_interval[[1L]]
+      )
+      # second count of date values coming from date within numeric
+      date_count <- date_count + sum(!is.na(y))
+      numeric_count <- sum(is.na(y))
+    }
+  } else {
+    tmp <- suppressWarnings(as.numeric(x))
+    numeric_count <- sum(!is.na(tmp))
   }
 
-  # --- get the proportion of logical values ---
-  are_logical   <- round((sum(is.logical(x)) / n_rows), 6L)
+  # get logical count
+  logicals <- toupper(x) == "TRUE" | toupper(x) == "FALSE"
+  logical_count <- sum(logicals)
 
-  # --- get the proportion of character values ---
-  are_character <- round((1.0 - (are_na + are_numeric +
-                                   are_date + are_logical)), 6L)
+  # get the character count
+  character_count <- initial_length -
+    (na_count + logical_count + numeric_count + date_count)
 
-  # --- return the output ---
-  return(c(are_na, are_numeric, are_date, are_character, are_logical))
-}
+  # transform into proportions
+  props <- round(
+    c(na_count, numeric_count, date_count, character_count, logical_count) /
+      initial_length, 4L)
```

Comment:
The guessing part only happens when `Date` values are identified from the parsing using `lubridate::parse_date_time()`. Otherwise, all numeric values are strictly considered as numeric. I think, the following could help users:

1. Add a note about this in the function description
2. Add the algorithm used within the function in the `details` section of the function documentation.

Reply:
With this in mind, I believe my question is still valid: should we do **any** guessing at all? If we don't have a strong reason to, I'd suggest we don't.

For example, if we have "20210107", should it be a date or a numeric? For a Covid-19 dataset, it could be either. It's probably safer to leave the final choice to the user.

Reply:
Here is another proposition from @Degoot-AM 

I would add a data type called `ambiguous`  and assigned to it non-differentiable entries and that would add the sum of raw to 1 (100%).